swap_server_client.go

```diff
@@ -202,16 +209,62 @@ func (s *grpcSwapServerClient) GetLoopInQuote(ctx context.Context,
 
 	rpcCtx, rpcCancel := context.WithTimeout(ctx, globalCallTimeout)
 	defer rpcCancel()
-	quoteResp, err := s.server.LoopInQuote(rpcCtx,
-		&looprpc.ServerLoopInQuoteRequest{
-			Amt:             uint64(amt),
-			ProtocolVersion: loopdb.CurrentRPCProtocolVersion,
-		},
-	)
+
+	quoteReq := &looprpc.ServerLoopInQuoteRequest{
+		Amt:             uint64(amt),
+		ProtocolVersion: loopdb.CurrentRPCProtocolVersion,
+	}
+
+	var probeHash lntypes.Hash
+	if loopdb.CurrentRPCProtocolVersion == looprpc.ProtocolVersion_MULTI_LOOP_IN_V2 {
+		// Generate random preimage.
+		var preimage lntypes.Preimage
+		if _, err := rand.Read(preimage[:]); err != nil {
+			log.Error("Cannot generate preimage")
+		}
+
+		probeHash := lntypes.Hash(sha256.Sum256(preimage[:]))
+
+		log.Infof("Creating quote probe invoice %v", probeHash)
+		probeInvoice, err := s.lnd.Invoices.AddHoldInvoice(
```

Comment:
why hold invoice? Also is 3600 expiry enough? I think the server tries for longer than that, or does the probe still work even if the invoice is expired.